### DISCUSSION TITLE
how to choose between various envs without writing to envs.toml

### DISCUSSION THREAD

**Discussion by piramiday** [python-poetry/poetry#9002] [Q&A] [answered]
hi all,

I have set up my project with three different environments (e.g. python 3.8, 3.9, 3.10) in a multi-user system, as a convenience for other users that do not know poetry.
I would like for other users to be able to run any environment of their choosing (between these 3), even at the same time (e.g. user A runs with python 3.8, user B runs with 3.9, etc).

is there a way for the `poetry run` command to be bound to one of these 3 envs?
at present, `poetry env use 3.X` will write into the `envs.toml` file that 3.X is the chosen env.
as a consequence, only people with write access to that TOML file are allowed to "switch env".

I guess I can replicate my repo folder 3 times and install 3 different envs, one python version per folder.
is there a better way to accomplish this? thank you!

**Comment by radoering** [comment, marked as answer]
I don't think there is a built-in way. Maybe, you can write a simple wrapper script that activates the appropriate environment (not via Poetry but like any other venv). If the venv is active, `poetry run` will run in this venv.
